src/mame/nmk/nmk16.cpp

```diff
@@ -3791,6 +3962,262 @@ static GFXDECODE_START( gfx_strahl )
+	m_spritegen->set_mask(0x3ff, 0x3ff);
+	m_spritegen->set_screen_size(60+380, 16+240);
+	m_spritegen->set_max_sprite_clock(448 * 263); // not verified?
+	m_spritegen->set_videoshift(60+32);
+}
+
+void nmk16_state::set_screen_hires(machine_config &config)
+{
+	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
+	m_screen->set_raw(XTAL(16'000'000)/2, 512, 28, 412, 278, 16, 240); // confirmed
+	m_screen->set_palette(m_palette);
+
+	NMK_16BIT_SPRITE(config, m_spritegen, XTAL(16'000'000)/2);
+	m_spritegen->set_screen_size(28+412, 16+240);
+	m_spritegen->set_max_sprite_clock(512 * 263); // not verified?
+	m_spritegen->set_videoshift(28+64);
+}
+
+/***************************************************************************
+
+                             Interrupt Handlers
+
+***************************************************************************/
+/*
+  While there are only 3 different H-timing PROMs (one for each screen resolution and ignoring the 4th bit used in
+  some games), multiple V-timings PROMs exists on games using this hardware.
+
+  The differences in these V-PROMs exists mainly because each game is configured to have own interrupt timing.
+  VSYNC, VBLANK, SPRITE-DMA timings are the same for all variants, though.
+
+  Summary of triggered IRQs on different games:
+
+  - IRQ1:
+    - At 68 and 196 scanlines on most games
+	- 'tharrier' only at 146 scanline
+	- 'vandyke', 'bioship' and 'blkheart' only at 102 scanline (also 'ddealer' outside this driver).
+	  'vandyke' permanently inhibits it by software.
+	- 'powerins' only at 16 scanline (looks like it's always inhibited by software)
+
+  - IRQ2:
+    - At 16 scanline on most games (VBIN = end of VBLANK = start of active video)
+	- 'tdragon', 'macross2', 'tdragon2' and 'raphero' lack it
+	- 'tharrier' at 54 scanline
+	- 'powerins' at 128 scanline (looks like it's always inhibited by software)
+
+  - IRQ3:
+    - Only triggered by 'powerins' at 90 and 166 scanlines (looks like it's always inhibited by software)
+
+  - IRQ4: (VBOUT)
+    - At 240 scanline for all games (VBOUT = start of VBLANK = end of active video)
+*/
+TIMER_DEVICE_CALLBACK_MEMBER(nmk16_state::nmk16_scanline)
+{
+//	const int SPRDMA_INDEX = 0;  // not used in emulation TODO: check if it could be used to trigger the sprite DMA instead of setting it 4 lines after VBOUT
+//	const int VSYNC_INDEX  = 1;  // not used in emulation
+//	const int VBLANK_INDEX = 2;  // not used in emulation
+//	const int NOT_USED     = 3;  // not used in emulation
+	const int IPL0_INDEX   = 4;
+	const int IPL1_INDEX   = 5;
+	const int IPL2_INDEX   = 6;
+	const int TRIGG_INDEX  = 7;
+
+	const int PROM_START_OFFSET = 0x75;  // previous entries are never addressed
+	const int PROM_FRAME_OFFSET = 0x0b;  // first 11 "used" entries (from 0x75 to 0x7f: 0xb entries) are prior to start of frame, which occurs on 0x80 address (128 entry)
+
+	u8 *prom = memregion("vtiming")->base();
+	int len = memregion("vtiming")->bytes();
```

Comment:
You should not be doing tag lookups on the hot path.  Use a `required_memory_region`.

Reply:
I use `optional_memory_region` instead, because some sets don't use it.